README.md

```diff
@@ -11,11 +13,135 @@ The modern, distributed and scalable implementation of a game inspired by Connec
+
+## Separation of Concerns
+The service names are all a reference to the popular anime `JoJo's Bizarre Adventure`.
+
+| Name       | Role            | Description                                                                   |
+| ---------- | --------------- | ----------------------------------------------------------------------------- |
+| Doppio     | Frontend        | Svelte-based web client                                                       |
+| Speedwagon | Load balancer   | Envoy-based load balancing reverse proxy                                      |
+| Joestar    | Scaling backend | Micronaut server for user authentication and caching                          |
+| Rohan      | Central backend | Micronaut server for central storage and processing of the game and its users |
+
+### Frontend
+The frontend is written in [Svelte](https://svelte.dev/).
+As it's main purpose is to display the current state of the board, we decided that frameworks such as Angular are overkill.
+
+### Load balancing
+We use [Envoy](https://www.envoyproxy.io/) as a reverse proxy to handle load balancing.
+We had to use Envoy as it's the only reverse proxy which currently supports grpc-web.
+
+### Backend
+The backend is split into two parts:
+* Scaling: Communicates directly with our Frontend, issues and validates JWT, caches game state and sends game updates to all clients
+* Central: Manages the actual game state, synchronizes requests and handles game logic, stores persistent information, such as user credentials and scores, in a json document
+
+Both backends use [Micronaut](https://micronaut.io/) with [Kotlin](https://kotlinlang.org/).
+
+### Storage
+We decided not to use a database but instead store the (very minimalistic) data in a JSON document.
+
+## Communication
+Bidirectional communication is enabled through [gRPC](https://grpc.io/).
+For example, this allows `Rohan` to send a game update event to all `Joestar` instances, which then forward them realtime to all `Doppio` clients.
+TODO:
+- gRPC:
+  - Authentication: Not used because of lack of support by Svelte ecosystem (overlap with [Symmetric JWT](#symmetric-jwt)?)?
+  - Bidirectional streaming and control flow: Updates only? How bidirectional is it?
+  - Blocking or non-blocking bindings: Relevant for programming (`suspend fun`s, `Promise`s + `async` + `await`)?
+  - Cancellation and timeouts: Relevant for programming (cancellation and timeout exceptions)?
+  - More?
+- JWT termination on Joestar servers
+- More?
+
+## Configuration
+TODO:
+- Locality
+- Processing
+- More?
+
+## Maintainability
+TODO:
+- Testing: 200+ unit tests
+- More?
+
+# Infrastructure
+
+## Docker setup
+TODO: Docker for everything
+
+## Reverse Proxy
+TODO:
+- Responsibilities (overlap with [Separation of concerns](#separation-of-concerns)?)
+- Load Balancing strategy
+- More?
+
+## CI/CD
+TODO:
+- GitHub actions
+- More?
+
+## PKI
+TODO:
+- Certbot
+- More?
+
+## Hosting
+TODO:
+- Linode
+- More?
```

Comment:
```suggestion
## Hosting

At the time of writing this, the latest stable version of ChaosConnect is deployed at [chaos.honegger.dev](https://chaos.honegger.dev/). We decided to use the cloud provider [Linode](https://www.linode.com/) because they provide fair pricing and a good free initial credit.

Setting up hosting was very easy, you just copy the docker-compose.yml file, replace the placeholder jwt-secret with a generated one and you're almost ready to start. If you don't already have a valid certificate around, you can easily generate one using [Certbot](https://certbot.eff.org/) using a command similar to the following and mounting them to their corresponding location within the Speedwagon container:

`sudo docker run -it --rm --name certbot -v "/etc/letsencrypt:/etc/letsencrypt" -v "/var/lib/letsencrypt:/var/lib/letsencrypt" -p 80:80 certbot/certbot certonly --standalone`
```

Reply:
Done